modules/opengl/src/buffer/bufferobject.cpp

```diff
@@ -245,8 +249,9 @@ void BufferObject::initialize(const void* data, GLsizeiptr sizeInBytes) {
     forEachObserver([](BufferObjectObserver* o) { o->onAfterBufferInitialization(); });
 }
 
-void BufferObject::upload(const void* data, GLsizeiptr sizeInBytes) {
-    if (sizeInBytes <= getSizeInBytes()) {
+void BufferObject::upload(const void* data, GLsizeiptr sizeInBytes, GrowPolicy policy) {
```

Comment:
How do we keep track of the last used size? can I ask for that? Seems we should have both a size and capacity here?